src/index.js

```diff
@@ -89,20 +90,26 @@ class Masq {
   _deleteSessionInfo () {
     window.localStorage.removeItem('userId')
     window.sessionStorage.removeItem('userId')
+    window.localStorage.removeItem('dataEncryptionKey')
+    window.sessionStorage.removeItem('dataEncryptionKey')
   }
 
-  _storeSessionInfo (stayConnected, userId) {
+  _storeSessionInfo (stayConnected, userId, dataEncryptionKey) {
     if (stayConnected) {
       window.localStorage.setItem('userId', userId)
+      window.localStorage.setItem('dataEncryptionKey', dataEncryptionKey)
     }
     window.sessionStorage.setItem('userId', userId)
+    window.sessionStorage.setItem('dataEncryptionKey', dataEncryptionKey)
   }
 
-  _loadSessionInfo () {
+  async _loadSessionInfo () {
     // If userId is in sesssion storage, use it and do not touch localStorage
     const sessionUserId = window.sessionStorage.getItem('userId')
-    if (sessionUserId) {
+    const sessionDataEncryptionKey = window.sessionStorage.getItem('dataEncryptionKey')
+    if (sessionUserId && sessionDataEncryptionKey) {
```

Comment:
Maybe we can use only check if sessionUserId is defined. In this case the key should be stored obviously right ?

Reply:
:+1:

Reply:
ok done